cli/ripsaw/commands/operator.py

```diff
@@ -0,0 +1,114 @@
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""CLI Commands for benchmark operations"""
+
+import os
+import subprocess
+import tempfile
+
+import click
+import git
+from ripsaw.clients.k8s import Cluster
+from ripsaw.util import logging
+
+logger = logging.get_logger(__name__)
+DEFAULT_REPO = "https://github.com/cloud-bulldozer/benchmark-operator"
+DEFAULT_BRANCH = "master"
+
+
+@click.group("operator")
+def operator_group():
+    """Passthrough function for Click CLI group, used as a decorator to declare functions for the CLI"""
+
+
+@operator_group.command("install")
+@click.option("--repo", help="Repo URL for benchmark-operator", show_default=True, default=DEFAULT_REPO)
+@click.option("--branch", help="Branch to checkout", show_default=True, default=DEFAULT_BRANCH)
+@click.option("--kubeconfig", help=" path to kubeconfig of cluster", default=None)
+def install(repo=DEFAULT_REPO, branch=DEFAULT_BRANCH, command="make deploy", kubeconfig=None):
+    """installs the operator from repo into specified cluster"""
+    click.echo(f"Installing Operator from repo {repo} and branch {branch}")
+    _perform_operator_action(repo, branch, command, kubeconfig)
+    wait_for_operator(kubeconfig=kubeconfig)
+    click.secho("Operator Running", fg="green")
+
+
+@operator_group.command("delete")
+@click.option("--repo", help="Repo URL for benchmark-operator", show_default=True, default=DEFAULT_REPO)
+@click.option("--branch", help="Branch to checkout", show_default=True, default=DEFAULT_BRANCH)
+@click.option("--kubeconfig", help="kubeconfig of cluster", default=None)
+def delete(repo=DEFAULT_REPO, branch=DEFAULT_BRANCH, command="make kustomize undeploy", kubeconfig=None):
+    """delete the operator from configured cluster"""
+    click.echo("Deleting Operator")
+    _perform_operator_action(repo, branch, command, kubeconfig)
+    click.secho("Operator Deleted", fg="green")
+
+
+def wait_for_operator(namespace="benchmark-operator", kubeconfig=None):
+    """wait for operator pods to be ready"""
+    cluster = Cluster(kubeconfig_path=kubeconfig)
+    label_selector = "control-plane=controller-manager"
+    cluster.wait_for_pods(label_selector, namespace, timeout=120)
+
+
+def _perform_operator_action(repo, branch, command, kubeconfig=None):
+    """run command from operator repo"""
+    shell_env = os.environ.copy()
+    if kubeconfig is not None:
+        shell_env["KUBECONFIG"] = kubeconfig
+    local_git_repo = _find_git_repo(os.getcwd())
+    result = None
+    if "benchmark-operator" in local_git_repo:
```

Comment:
Does this override the default repo/branch based on the git repo & branch if you're in one locally as it's not reflected in that initial "Installing Operator from repo..." print statement?

Reply:
When not installing/deleting outside of a git repo it's not too happy:

```
$ ripsaw operator install
Installing Operator from repo https://github.com/cloud-bulldozer/benchmark-operator and branch master
Traceback (most recent call last):
  File "/home/user/go/src/github.com/cloud-bulldozer/benchmark-operator/venv/bin/ripsaw", line 33, in <module>
    sys.exit(load_entry_point('ripsaw-cli==0.0.1', 'console_scripts', 'ripsaw')())
  File "/home/user/go/src/github.com/cloud-bulldozer/benchmark-operator/venv/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/home/user/go/src/github.com/cloud-bulldozer/benchmark-operator/venv/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/home/user/go/src/github.com/cloud-bulldozer/benchmark-operator/venv/lib/python3.9/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/user/go/src/github.com/cloud-bulldozer/benchmark-operator/venv/lib/python3.9/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/user/go/src/github.com/cloud-bulldozer/benchmark-operator/venv/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/user/go/src/github.com/cloud-bulldozer/benchmark-operator/venv/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/home/user/go/src/github.com/cloud-bulldozer/benchmark-operator/venv/lib/python3.9/site-packages/ripsaw/commands/operator.py", line 42, in install
    _perform_operator_action(repo, branch, command, kubeconfig)
  File "/home/user/go/src/github.com/cloud-bulldozer/benchmark-operator/venv/lib/python3.9/site-packages/ripsaw/commands/operator.py", line 72, in _perform_operator_action
    if "benchmark-operator" in local_git_repo:
TypeError: argument of type 'NoneType' is not iterable
```

Reply:
yep the conditional needed to check to make sure `local_git_repo is not None`. Just pushed a fix

Also fixed the logs around telling you what repo/branch you're on.

Reply:
In the case where the user provides `ripsaw operator install` a `--repo` or `--branch` inside of the git repo, can we have the CLI args override even the git repo?

Reply:
the only real way to do that is to add a flag to force using a remote repo, which I just added as `--force-remote`. This has the added benefit of being able to force using the remote repo with the default repo/branch